### PR TITLE
Labels feature

### PR DESCRIPTION
**Pull request by kiranbayram**
Opening this PR to make it transparent that i'm working on this feature

### CODE REVIEW

**Comment by kiranbayram** [comment]
Hello @fthomas , I wanted to ask you before continuing if I should add a `labels` field to `Update` model or should I keep the labels outside and write a `LabelAlg` which returns labels for an `Update` ? What do you think ?

**Comment by fthomas** [comment]
@kiranbayram How would a potential `LabelAlg` determine a label for an `Update`? I think at the time scala-steward found `Update`s for a project it has forgotten if it is an update for a library dependency, a test dependency, or for a a plugin and that information can't be recreated easily. So I would think that scala-steward need to somehow retain that info when it determines `Update`s for a project.

**Comment by kiranbayram** [comment]
Then it sounds like we need to add labels field to Update model and use it while creating PR body

File: modules/core/src/main/scala/org/scalasteward/core/model/Label.scala [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.scalasteward.core.model
+import io.circe.{Decoder, Encoder}
+
+sealed trait Label {
```
Comment (Fristi) [inline modules/core/src/main/scala/org/scalasteward/core/model/Label.scala]: You could make this a `sealed abstract class Label(val name: String)` which save you some lines below

**Comment by kiranbayram** [comment]
@Fristi Thanks for the review. I think, it might be useful to automatically identify a dependency type but maybe with another PR.

**Comment by kiranbayram** [comment]
Hello @fthomas, could you please review ?

**Comment by fthomas** [comment]
Do I understand it correctly that this PR basically allows users to define a mapping from dependencies to labels in `.scala-steward.conf` which scala-steward then uses to label PRs of matching dependencies?

**Comment by kiranbayram** [comment]
Yes, that is right

**Comment by kiranbayram** [comment]
I thought this was what you suggested in your comment on the issue

**Comment by fthomas** [comment]
Sorry for the confusion, @kiranbayram. My understanding of #404 was this: scala-steward knows when it gets updates from sbt-updates if a dependency is a normal or test dependency (this is stored in `Update.Single.configurations`) and it could know which updates are for build dependencies and plugins. With this info we could label them automatically without the need for any configuration. With customisation of the labels I meant to configure the names of those labels via the config file.

**Comment by kiranbayram** [comment]
Hmm, sorry from my side too. I should probably have communicated what i wanted to do before doing it 😊 I can try to adjust the PR once i get some free time

**Comment by kiranbayram** [comment]
I think it'd be easier to implement this feature from scratch than updating this branch, that's why i'm going to close this PR.

**Comment by MPV** [comment]
Did anyone (@kiranbayram) give this another try? ❤️

**Comment by MPV** [comment]
Or was https://github.com/fthomas/scala-steward/pull/755 the merged solution for this? ❔👍

**Comment by exoego** [comment]
If you mean #404 for "this", yes, I think the merged PR #755 was for it and superceded this PR.

**Comment by MPV** [comment]
Ah yes, thanks @exoego ❤️ 

I opened #1471 to discuss also adding proper labels (not just into PR description).

**Pull request closed without merging**